pkg/ottl/ottlfuncs/func_parse_severity.go

```diff
@@ -0,0 +1,173 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+const (
+	// hTTP2xx is a special key that is represents a range from 200 to 299
+	hTTP2xx = "2xx"
+
+	// hTTP3xx is a special key that is represents a range from 300 to 399
+	hTTP3xx = "3xx"
+
+	// hTTP4xx is a special key that is represents a range from 400 to 499
+	hTTP4xx = "4xx"
+
+	// hTTP5xx is a special key that is represents a range from 500 to 599
+	hTTP5xx = "5xx"
+
+	minKey = "min"
+	maxKey = "max"
+)
+
+type ParseSeverityArguments[K any] struct {
+	Target  ottl.Getter[K]
+	Mapping ottl.PMapGetter[K]
+}
+
+func NewParseSeverityFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("ParseSeverity", &ParseSeverityArguments[K]{}, createParseSeverityFunction[K])
+}
+
+func createParseSeverityFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ParseSeverityArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("ParseSeverityFactory args must be of type *ParseSeverityArguments[K")
+	}
+
+	return parseSeverity[K](args.Target, args.Mapping), nil
+}
+
+func parseSeverity[K any](target ottl.Getter[K], mapping ottl.PMapGetter[K]) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		severityMap, err := mapping.Get(ctx, tCtx)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get severity mapping: %w", err)
+		}
+
+		value, err := target.Get(ctx, tCtx)
+		if err != nil {
+			return nil, fmt.Errorf("could not get log level: %w", err)
+		}
+
+		logLevel, err := evaluateSeverity(value, severityMap.AsRaw())
```

Comment:
Is there anything from stanza's severity parser we could borrow here? It would be nice if they shared logic, similar to how we did the time parsing stuff

Reply:
@bacherfl Ok your solution sounds good to me. If we add some static lookup feature in the future (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/39296) maybe we can revisit.